File: 프로그래머스/lv1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    answer = 0
    if num == 1:
        answer = 0
    else:
        while num != 1:
            if num % 2 == 0:
                #짝수
                num = num // 2
                answer += 1
            elif num % 2 == 1:
                #홀수
                num = (num * 3) + 1
                answer += 1
            if answer == 500:
                answer = -1
                return answer
    return answer

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
//입력된 숫자가 짝수면 2로 나누기
//입력된 수가 홀수면 3을 곱하고 1을 더하기
//결과로 나온 수에 같은 작업을 1이 될 때까지 반복
class Solution {
    public int solution(int num) {
        int answer = num;
        int count = 0;
        while(answer != 1) {
            if(answer % 2 == 0) {
                //짝수
                answer = answer / 2;
            }
            else if(answer % 2 == 1) {
                //홀수
                answer = (answer * 3) + 1;
            }
            
            if(count >= 500) {
                count = -1;
                break;
            }
            count += 1;      
        }
        return count;
    }
}